contrib/gitian-build.sh

```diff
@@ -1,406 +1,389 @@
-        --no-commit)
-            commitFiles=false
+        # URL
+        -u)
+            if [ -n "$2" ]; then
+                url=$2
+                shift
+            else
+                echo 'Error: "-u" requires an argument'
+                exit 1
+            fi
             ;;
         # Setup
         --setup)
             setup=true
             ;;
-	*)               # Default case: If no more options then break out of the loop.
-             break
+        --test)
+            test=true
+            ;;
+        *) # Default case: If no more options then break out of the loop.
+            break ;;
     esac
     shift
 done
 
-# Set up LXC
-if [[ $lxc = true ]]
-then
-    export USE_LXC=1
-    export LXC_BRIDGE=br0
+# Download specific file and verify hash
+function download_file () {
+    local filename=$(basename $1)
+
+    if [ ! -f $filename ]; then
+        wget $1
+    fi
+
+    # Verify file signature
+    echo "$2 $filename" | sha256sum -c --status
+
+    if [ $? != 0 ]; then
+        echo "$scriptName: Signature for $filename don't match."
+        exit 1
+    fi
+}
+
+function move_build_files() {
+    find build/out -type f -exec mv '{}' $outputDir/dogecoin-binaries/${VERSION}/ \;
+}
+
+function download_descriptor() {
+    if [[ ! $1 == 'test' ]]; then
+        uri="${url/github.com/raw.githubusercontent.com}"/"$2"/contrib/gitian-descriptors/gitian-"$1".yml
+        echo "Downloading descriptor ${1} ${uri}"
+        wget $uri -O gitian-"$1".yml || exit 1
+    else
+        # CI tests
+        cp ../ci/descriptor/"$1".yml gitian-"$1".yml || exit 1
+    fi
+}
+
+### Test configuration ###
+
+if [[ $test == true ]]; then
+    if [[ $commit == true ]]; then
+        VERSION="f80bfe9068ac1a0619d48dad0d268894d926941e"
+    else
+        VERSION="1.14.3"
+    fi
+    DESCRIPTORS=('test')
+    SIGN_DESCRIPTORS=()
+    COMMIT=$VERSION
+    SIGNER="signer"
+    SHA256SUM="0d519f6ade0e601617a7c44b764eeae35a8784070c3e44f542011956f1743459"
 fi
 
-# Check for OSX SDK
-if [[ ! -e "gitian-builder/inputs/MacOSX10.11.sdk.tar.gz" && $osx == true ]]
-then
-    echo "Cannot build for OSX, SDK does not exist. Will build for other OSes"
-    osx=false
+### Arguments checks ####
+
+echo "Using ${proc} CPU and ${mem} RAM"
+
+# Control the selection of a single virtualisation software
+if [ $(($USE_LXC + $USE_DOCKER)) -ge 2 ]; then
+    echo "$scriptName: Specify a single virtualisation solution between Docker, LXC or KVM."
+    exit
 fi
 
-# Get signer
-if [[ -n"$1" ]]
-then
-    SIGNER=$1
-    shift
+if [ -n "$SIGNER" ]; then
+    echo "Testing GPG Keys available..."
+    result=$(gpg --list-secret-keys --keyid-format=long | grep sec | grep -v revoked | grep "" -c)
+    if [[ $result == "" ]]; then
+        echo "No GPG keys available..."
+        echo "Please follow this documentation: https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification/generating-a-new-gpg-key"
+        exit 1
```

Comment:
This is incompatible with detached signing. Could you make it so that detached signing is possible again (i.e. create all the outputs for signing, but simply don't sign?

Reply:
Fine, I'm going to remove the entire check. The current verification check if some keys exists, do not verify if it matches the signer. I was able to perform detached signing with it because I had some available keys.